validation/src/lib.rs

```diff
@@ -186,9 +186,11 @@ pub trait Network {
 #[derive(Debug, Clone, Default)]
 pub struct GroupInfo {
 	/// Authorities meant to check validity of candidates.
-	pub validity_guarantors: HashSet<SessionKey>,
+	validity_guarantors: HashSet<SessionKey>,
+	/// Mapping from validator index to public key.
+	index_mapping: HashMap<ValidatorIndex, SessionKey>,
```

Comment:
I find it a bit strange to reproduce this into every `GroupInfo`. Why not have this for the table as a whole?